test/test_test.py

```diff
@@ -492,11 +492,39 @@ def test_reciprocal_blades(self, g):
                     assert ga.scalar_product(b1, rb2).simplify() == S.Zero
 
     def test_metric_collect(self):
-        ga = Ga('e*1|2', g=[1, 1, 1])
+        ga = Ga('e*1|2', g=[1, 1])
         e1, e2 = ga.basis
 
         assert metric.collect(2*e1 + e2, [e1]) == 2*e1 + e2
 
+    def test_dual_mode(self):
+        ga, e1, e2 = Ga.build('e*1|2', g=[1, 1])
+
+        default = Ga.dual_mode_value
+        assert default == 'I+'
+        with pytest.raises(ValueError):
+            Ga.dual_mode('illegal')
+
+        d_default = e1.dual()
+
+        # note: this is a global setting, so we have to make sure we put it back
```

Comment:
I believe this should be GA specific so not a global one. There could be G(3), PGA, CGA mixed together in a program and they might want to use different dual mode? I don't know, I used to believe this is only a flavor which is decided by the coder only so it could be global to the program.......

Reply:
I completely agree, this behavior is bad at the moment. Something for 0.6.0.